samples/bluetooth/channel_sounding_ras_initiator/src/main.c

```diff
@@ -0,0 +1,569 @@
+	}
+}
+
+static void scan_filter_match(struct bt_scan_device_info *device_info,
+			      struct bt_scan_filter_match *filter_match, bool connectable)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(device_info->recv_info->addr, addr, sizeof(addr));
+
+	printk("Filters matched. Address: %s connectable: %d\n", addr, connectable);
+}
+
+static void scan_connecting_error(struct bt_scan_device_info *device_info)
+{
+	int err;
+
+	printk("Connecting failed, restarting scanning\n");
+
+	err = bt_scan_start(BT_SCAN_TYPE_SCAN_PASSIVE);
+	if (err) {
+		printk("Failed to restart scanning (err %i)\n", err);
+		return;
+	}
+}
+
+static void scan_connecting(struct bt_scan_device_info *device_info, struct bt_conn *conn)
+{
+	printk("Connecting\n");
+}
+
+BT_SCAN_CB_INIT(scan_cb, scan_filter_match, NULL, scan_connecting_error, scan_connecting);
+
+static int scan_init(void)
+{
+	int err;
+
+	struct bt_scan_init_param param = {
+		.scan_param = NULL, .conn_param = BT_LE_CONN_PARAM_DEFAULT, .connect_if_match = 1};
+
+	bt_scan_init(&param);
+	bt_scan_cb_register(&scan_cb);
+
+	err = bt_scan_filter_add(BT_SCAN_FILTER_TYPE_UUID, BT_UUID_RANGING_SERVICE);
+	if (err) {
+		printk("Scanning filters cannot be set (err %d)\n", err);
+		return err;
+	}
+
+	err = bt_scan_filter_enable(BT_SCAN_UUID_FILTER, false);
+	if (err) {
+		printk("Filters cannot be turned on (err %d)\n", err);
+		return err;
+	}
+
+	return 0;
+}
+
+BT_CONN_CB_DEFINE(conn_cb) = {
+	.connected = connected_cb,
+	.disconnected = disconnected_cb,
+	.le_param_req = le_param_req,
+	.security_changed = security_changed,
+	.le_cs_remote_capabilities_available = remote_capabilities_cb,
+	.le_cs_config_created = config_created_cb,
+	.le_cs_security_enabled = security_enabled_cb,
+	.le_cs_procedure_enabled = procedure_enabled_cb,
+	.le_cs_subevent_data_available = subevent_result_cb,
+};
+
+int main(void)
+{
+	int err;
+
+	printk("Starting Channel Sounding Initiator Demo\n");
+
+	dk_leds_init();
+
+	err = bt_enable(NULL);
+	if (err) {
+		printk("Bluetooth init failed (err %d)\n", err);
+		return 0;
+	}
+
+	err = scan_init();
+	if (err) {
+		printk("Scan init failed (err %d)\n", err);
+		return 0;
+	}
+
+	err = bt_scan_start(BT_SCAN_TYPE_SCAN_PASSIVE);
+	if (err) {
+		printk("Scanning failed to start (err %i)\n", err);
+		return 0;
+	}
+
+	while (true) {
+		err = k_sem_take(&sem_connected, K_FOREVER);
+		__ASSERT_NO_MSG(!err);
+
+		reconnecting = false;
+
+		err = bt_conn_set_security(connection, BT_SECURITY_L2);
+		if (err) {
+			printk("Failed to encrypt connection (err %d)\n", err);
+			return 0;
+		}
+
+		err = k_sem_take(&sem_security, K_SECONDS(5));
+		if (err) {
+			printk("Timeout waiting for pairing (err %d)\n", err);
+			return 0;
+		}
+
+		static struct bt_gatt_exchange_params mtu_exchange_params = {
+			.func = mtu_exchange_cb};
+
+		err = bt_gatt_exchange_mtu(connection, &mtu_exchange_params);
+		if (err) {
+			printk("MTU exchange failed (err %d)\n", err);
```

Comment:
I don't think it's worth handling the return code here, the only error code you could get is EALREADY which means the sem_mtu_exchange_done semaphore would have already been given in the MTU exchange callback.